packages/url_launcher/url_launcher_ios/example/ios/RunnerTests/URLLauncherTests.swift

```diff
@@ -81,22 +68,15 @@ final class URLLauncherTests: XCTestCase {
 
   func testLaunchFailureWithInvalidURL() {
     let expectation = XCTestExpectation(description: "completion called")
-
-    createPlugin().launchURL("urls can't have spaces", universalLinksOnly: false) { result, error in
-      if (error == nil) {
+    createPlugin().launchUrl(url: "urls can't have spaces", universalLinksOnly: false) { result in
+      switch result {
+      case .success(let details):
         // When linking against the iOS 17 SDK or later, NSURL uses a lenient parser, and won't
         // fail to parse URLs, so the test must allow for either outcome.
-        XCTAssertNotNil(result)
-        XCTAssertFalse(result?.boolValue ?? true)
-        XCTAssertNil(error)
-      } else {
-        XCTAssertNil(result)
-        XCTAssertNotNil(error)
-        XCTAssertEqual(error?.code, "argument_error")
-        XCTAssertEqual(error?.message, "Unable to parse URL")
-        XCTAssertEqual(error?.details as? String, "Provided URL: urls can't have spaces")
+        XCTAssertTrue(details == .failedToLoad || details == .invalidUrl)
```

Comment:
which case will it actually be?

Reply:
It depends on the iOS version the tests are being run on. I think @stuartmorgan can explain it better, but on iOS 17 we should get .failedToLoad, and on older versions, we should get .invalidUrl

Reply:
Can you do "if iOS 17 else".

Reply:
I realized there's a better option; I've added a utility method in the test file that checks the NSURL behavior directly, and then switches based on that in the tests.